autogpt_platform/frontend/src/components/agptui/Wallet.tsx

```diff
@@ -0,0 +1,114 @@
+"use client";
+
+import useCredits from "@/hooks/useCredits";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { X } from "lucide-react";
+import { PopoverClose } from "@radix-ui/react-popover";
+import { TaskGroups } from "../onboarding/WalletTaskGroups";
+import { ScrollArea } from "../ui/scroll-area";
+import { useOnboarding } from "../onboarding/onboarding-provider";
+import { useCallback, useEffect, useRef } from "react";
+import { cn } from "@/lib/utils";
+import * as party from "party-js";
+
+export default function Wallet() {
+  const { credits, formatCredits, fetchCredits } = useCredits({
```

Comment:
I have a feeling you could pop this into v0 and say use modern nextjs best practices and get a server component out